app/publisher/publisher.py

```diff
@@ -0,0 +1,37 @@
+import google
+from google.cloud.pubsub_v1 import PublisherClient
+from google.cloud.pubsub_v1.futures import Future
+from structlog import get_logger
+
+from app.publisher.publication_failed import PublicationFailed
+
+logger = get_logger(__name__)
+
+
+class PubSub(PublisherClient):
+    def __init__(self, topic_id):
+        super().__init__()
+
+        _, project_id = google.auth.default()
+        self.topic_path = super().topic_path(project_id, topic_id)
+
+    def _publish(self, message):
```

Comment:
Just for readability and my preference is to put private methods after public. It also makes the class flow better when reading, if I am following the code naturally publish_and_resolve comes before _publish

Reply:
My personal preference is to define all underscore/Dunder methods first. Obviously in Python in most cases the order you define methods doesn't matter but I did it like this as it is a common convention to define methods before they are invoked.